### PR TITLE
Use variables to replace hard-coding value in clamp function

### LINKED ISSUE #151: Use variables names to replace hard-coding value in clamp function
This issue comes from one [PR comment](https://github.com/1QB-Information-Technologies/ccvm/pull/141#discussion_r1448027749).

- The values `0` and `1` in the clamp function should be general values named `lower_clamp` and `upper_clamp`, which receive their values at the input with the default values set to `lower_clamp=0.0` and `upper_clamp=1.0`.

- This also applies to all post processors.

### PR DESCRIPTION
**Pull request by Estella-Wang-1qbit**
- Enhanced post processor clamp function for greater flexibility by replacing hard-coded values with user-defined variables (`lower_clamp` and `upper_clamp`).
- Closes #151

### CODE REVIEW

File: ccvm_simulators/post_processor/tests/test_adam.py [hunk 1]
```diff
@@ -39,6 +39,22 @@ def test_postprocess_valid(self):
         error_message = "post_processing time must be greater than 0"
         self.assertGreater(self.post_processor.pp_time, 0, error_message)
 
+    def test_postprocess_default_upper_lower_clamp(self):
```
Comment (KatieO4t1qbit) [inline ccvm_simulators/post_processor/tests/test_adam.py:42]:
```suggestion
    def test_postprocess_custom_upper_lower_clamp(self):
```

Comment (KatieO4t1qbit) [reply ccvm_simulators/post_processor/tests/test_adam.py:42]: Same comment for all of the test files

File: ccvm_simulators/post_processor/tests/test_lbfgs.py [hunk 2]
```diff
@@ -37,6 +37,22 @@ def test_postprocess_valid(self):
         error_message = "post_processing time must be greater than 0"
         self.assertGreater(self.post_processor.pp_time, 0, error_message)
 
+    def test_postprocess_default_upper_lower_clamp(self):
```
Comment (KatieO4t1qbit) [inline ccvm_simulators/post_processor/tests/test_lbfgs.py:40]: Don't forget to rename the `test_postprocess_valid` to `test_postprocess_default_values` in this file, too :)

**Review by KatieO4t1qbit — APPROVED** [review]
Looks great, Estella!!
Only a couple minor comments.

File: ccvm_simulators/post_processor/tests/test_adam.py (hunk 1 above)
Comment (Estella-Wang-1qbit) [reply ccvm_simulators/post_processor/tests/test_adam.py]: Done.

File: ccvm_simulators/post_processor/tests/test_lbfgs.py (hunk 2 above)
Comment (Estella-Wang-1qbit) [reply ccvm_simulators/post_processor/tests/test_lbfgs.py]: Done.

**Review by FarhadK-1QBit — APPROVED** [review]
Thank you for this change, Estella! It looks great to me!

**Pull request merged**